src/xrpld/rpc/detail/TransactionSign.cpp

```diff
@@ -531,10 +531,38 @@ transactionPreProcessImpl(
         if (!signingArgs.isMultiSigning())
         {
             // Make sure the account and secret belong together.
-            auto const err = acctMatchesPubKey(sle, srcAddressID, pk);
+            if (tx_json.isMember(sfDelegate.jsonName))
+            {
+                // Delegated transaction
+                auto const ptrDelegatedAddressID = parseBase58<AccountID>(
+                    tx_json[sfDelegate.jsonName].asString());
```

Comment:
Here also it should be checked that the field is a string. If the field is an object or array, an exception will be thrown.